src/serialize.ts

```diff
@@ -253,6 +253,25 @@ export function decodeString(
   return String(str);
 }
 
+/**
+ * Decodes an enum value while safely handling null and undefined values.
+ *
+ * If an array is provided, only the first entry is used.
+ *
+ * @param {String} input the encoded string
+ * @param {String[]} enumValues allowed enum values
+ * @return {String} the string value from enumValues
+ */
+export function decodeEnum<T extends string>(
+  input: string | (string | null)[] | null | undefined,
+  enumValues: T[]
+): T | null | undefined {
+  const str = decodeString(input)
+  if (str == null) return str
+  const isEnumMember = (str: string): str is T => enumValues.includes(str as any)
+  return isEnumMember(str) ? str : undefined
```

Comment:
Can we avoid creating this intermediate function by writing the code this way?

```ts
  if (enumValues.includes(str as any)) {
    return str as T
  }
  return undefined
```

Reply:
Sure! Done